### PR TITLE
Add territory class

### PR DESCRIPTION
**Pull request by BenSna**

### CODE REVIEW

File: test_territory.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        territory = Territory('test', 0, 'continent', [], 'player1', 0, [], [])
+        territory.gain_troops(10)
+        self.assertEqual(territory.quantity, 10)
+
+    def test_lose_troops(self):
+        territory = Territory('test', 0, 'continent', [], 'player1', 10, [], [])
+        territory.lose_troops(10)
+        self.assertEqual(territory.quantity, 0)
+
+    def test_losing_more_troops_than_available_raises_exception(self):
```
Comment (dan-nash) [inline test_territory.py:17]: I added this test, which fails, needs to be fixed

File: territory.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+class Territory:
+    def __init__(self, name, index, continent, adjacent, controller, quantity, location, corners):
+        self.controller = controller
+        self.quantity = quantity
```
Comment (dan-nash) [inline territory.py:4]: don't think "quantity" is a good variable name, what about "troop_count"

File: territory.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        self.continent = continent
+        self.index = index
+        self.adjacent = adjacent
```
Comment (dan-nash) [inline territory.py:7]: again be explicit with var names, "adjacent_territories"

File: territory.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        self.name = name
+        self.location = location
+        self.corners = corners
```
Comment (dan-nash) [inline territory.py:10]: I don't think we need location and corners yet, as that's for the UI stuff

**Review by dan-nash — CHANGES_REQUESTED** [review]

File: territory.py (hunk 2 above)
Comment (BenSna) [reply territory.py]: fixed

File: territory.py (hunk 3 above)
Comment (BenSna) [reply territory.py]: fixed

File: test_territory.py (hunk 1 above)
Comment (BenSna) [reply test_territory.py:17]: fixed

File: territory.py (hunk 4 above)
Comment (dan-nash) [reply territory.py:10]: Yeah fair enough let's leave it in then

**Comment by dan-nash** [comment]
Can you add a couple more tests and error checks so gain troops and lose troops can't take negative numbers? then it's ready to go :)
